lizmap/modules/lizmap/lib/Project/ProjectMetadata.php

```diff
@@ -211,6 +211,57 @@ public function getLizmapWebCLientTargetVersion()
         return $this->data['lizmapWebClientTargetVersion'];
     }
 
+    /**
+     * If the QGIS desktop needs an update of the plugin.
+     * The check is done only if the project has been edited recently compare to the date of the Lizmap Web Client.
+     *
+     * @return bool If the plugin should be updated in QGIS Desktop
+     */
+    public function qgisLizmapPluginUpdateNeeded()
+    {
+        $projectInfos = \Jelix\Core\Infos\AppInfos::load();
+        $releaseVersion = $projectInfos->version;
+        if (substr_count($releaseVersion, '-') <= 1 && substr_count($releaseVersion, '.') == 2) {
+            // We are in dev mode : version is 3.8.0-pre for instance
+            // Dates on QGS files are not totally correct, due to git, we do not process this check for now
+            // A non official package is like 3.6.11-pre.6402, so the check will be done
+            return false;
+        }
+
+        // Later, we can maybe use the date in the XML file, reading the first 3 lines
+        $projectDate = date($this->getFileTime());
+
+        $releaseDate = \jApp::config()->minimumRequiredVersion['lizmapDesktopPluginDate'];
+
+        if ($projectDate < strtotime($releaseDate)) {
+            // Project file date is older than internal release date, we do nothing
+            // Project can stay on the server without any update
+            return false;
+        }
+
+        // Project file is newer than the release date
+        // We check against the hard coded version
+        $recommendedVersion = \jApp::config()->minimumRequiredVersion['lizmapDesktopPlugin'];
+
+        $projectVersion = $this->getLizmapPluginVersion();
+        if (!is_numeric($projectVersion)) {
+            // It shouldn't happen to much, but before the version was like "master".
```

Comment:
Is it for old plugin that return version as human readable string ?
should'nt return true ?

Reply:
I will rephrase.
`master` is still written in the git repository : https://github.com/3liz/lizmap-plugin/blob/master/lizmap/metadata.txt#L17
It's rare, but it can happen if we are running QGIS in a docker container for instance (GIT on the host to fetch the git repository and then running QGIS desktop in a docker container (so without git), the version will be `master` in the CFG file. I will keep this in mind.